georocket-client-api/src/main/java/io/georocket/client/StoreClient.java

```diff
@@ -170,28 +199,32 @@ protected String prepareQuery(String query, String layer) {
    */
   public WriteStream<Buffer> startImport(String layer, Collection<String> tags,
       Optional<Long> size, Handler<AsyncResult<Void>> handler) {
-    String path = "/store";
-    
-    if (layer != null && !layer.isEmpty()) {
-      if (!layer.endsWith("/")) {
-        layer += "/";
-      }
-      if (!layer.startsWith("/")) {
-        layer = "/" + layer;
-      }
-      path += layer;
-    }
-    
-    if (tags != null && !tags.isEmpty()) {
-      path += "?tags=" + String.join(",", tags);
-    }
-    
+
+    String path = prepareImport(layer, tags);
     HttpClientRequest request = client.post(path);
-    
+    return requestOptions(request, size, handler);
+  }
+
+  /**
+   * Set options for HTTP requests. Because some requests require
+   * special options, the option setting is outsourced to this
+   * method.
+   *
+   * @param request request the options are created for
+   * @param size size of the data to be sent in bytes (optional)
+   * @param handler a handler that will be called when the data has been
+   * imported by the GeoRocket server
+   * @return same {@link HttpClientRequest} as given as parameter but with options set
+   */
+  protected HttpClientRequest requestOptions(HttpClientRequest request, Optional<Long> size, Handler<AsyncResult<Void>> handler) {
```

Comment:
Line is very long. It almost doesn't fit here in the diff view. Please wrap it like in `startImport`.